docs/src/background/infeasibility_certificates.md

```diff
@@ -0,0 +1,154 @@
+\\
+& \;\;\text{s.t.} & a_0 + \sum_{i=1}^m A_i^\top y_i & = 0
+\\
+& & y_i & \in \mathcal{C}_i^* & i = 1 \ldots m.
+\end{align}
+```
+
+## Unbounded models
+
+If a model is unbounded, two things are true:
+ 1. there exists a feasible primal solution
+ 2. the dual is infeasible.
+
+A feasible primal solution---if one exists---can be obtained by setting
+[`ObjectiveSense`](@ref) to `FEASIBILITY_SENSE` and then optimizing. Therefore,
+most solvers terminate after they prove the dual is infeasible via a certificate
+of dual infeasibility, but _before_ they have found a feasible primal solution.
+This is also the reason that MathOptInterface defines the `DUAL_INFEASIBLE`
+status instead of `UNBOUNDED`.
+
+A certificate of dual infeasibility is an improving ray of the primal problem.
+That is, there exists some vector ``d`` such that for all ``\eta > 0``:
+```math
+A_i (x + \eta d) + b_i \in \mathcal{C}_i,\ \ i = 1 \ldots m,
+```
+and (for minimization problems):
+```math
+a_0^\top (x + \eta d) + b_0 < a_0^\top x + b_0,
+```
+for any feasible point ``x``. The latter simplifies to ``a_0^\top d < 0``. For
+maximization problems, the inequality is reversed, so that ``a_0^\top d > 0``.
+
+If the solver has found a certificate of dual infeasibility:
+
+ * [`TerminationStatus`](@ref) must be `DUAL_INFEASIBLE`
+ * [`PrimalStatus`](@ref) must be `INFEASIBILITY_CERTIFICATE`
+ * [`VariablePrimal`](@ref) must be the corresponding value of ``d``
+ * [`ObjectiveValue`](@ref) must be the value ``a_0^\top d``. Note that this is
+   the value of the objective function at `d`, ignoring the constant `b_0`.
+
+!!! note
+    The choice of whether to scale the ray ``d`` to have magnitude `1` is left
+    to the solver.
+
+## Infeasible models
+
+A certificate of primal infeasibility is an improving ray of the dual problem.
+However, because infeasibility is independent of the objective function, we
+first homogenize the primal problem by removing its objective.
+
+For a minimization problem, a dual improving ray is some vector ``d`` such that
+for all ``\eta > 0``:
+```math
+\begin{align}
+\sum_{i=1}^m -A_i^\top (y_i + \eta d_i) & = 0 \\
+(y_i + \eta d_i) & \in \mathcal{C}_i^* & i = 1 \ldots m,
+\end{align}
+```
+and:
+```math
+-\sum_{i=1}^m b_i^\top (y_i + \eta d_i) > -\sum_{i=1}^m b_i^\top y_i,
+```
+for any feasible dual solution ``y``. The latter simplifies to
+``-\sum_{i=1}^m b_i^\top d_i > 0``. For a maximization problem, the inequality
+is ``\sum_{i=1}^m b_i^\top d_i < 0``. (Note that these are the same inequality,
+modulo a `-` sign.)
+
+If the solver has found a certificate of infeasibility:
+
+ * [`TerminationStatus`](@ref) must be `INFEASIBLE`
+ * [`DualStatus`](@ref) must be `INFEASIBILITY_CERTIFICATE`
+ * [`ConstraintDual`](@ref) must be the corresponding value of ``d``
+ * [`DualObjectiveValue`](@ref) must be the value
+   ``-\sum_{i=1}^m b_i^\top d_i`` for minimization problems and
+   ``\sum_{i=1}^m b_i^\top d_i`` for maximization problems.
```

Comment:
There's also a question about what to do here. Currently, the dual fallback returns different signs, so that's what I've left it as.

Reply:
I've convinced myself that this is correct.